airbyte-integrations/connectors/destination-postgres/src/test/java/io/airbyte/integrations/destination/postgres/PostgresDestinationTest.java

```diff
@@ -132,6 +140,72 @@ void testDefaultParamsWithSSL() {
     assertEquals(SSL_JDBC_PARAMETERS, defaultProperties);
   }
 
+  @Test
+  void testCheckIncorrectPasswordFailure() {
+    var config = buildConfigNoJdbcParameters();
+    ((ObjectNode) config).put(JdbcUtils.PASSWORD_KEY, "fake");
+    var destination = new PostgresDestination();
+    var actual = destination.check(config);
+    assertTrue(actual.getMessage().contains("State code: 28P01;"));
+  }
+
+  @Test
+  public void testCheckIncorrectUsernameFailure() {
+    var config = buildConfigNoJdbcParameters();
+    ((ObjectNode) config).put(JdbcUtils.USERNAME_KEY, "");
+    var destination = new PostgresDestination();
+    final AirbyteConnectionStatus actual = destination.check(config);
+    assertTrue(actual.getMessage().contains("State code: 28P01;"));
+  }
+
+  @Test
+  public void testCheckIncorrectHostFailure() {
+    var config = buildConfigNoJdbcParameters();
+    ((ObjectNode) config).put(JdbcUtils.HOST_KEY, "localhost2");
+    var destination = new PostgresDestination();
+    final AirbyteConnectionStatus actual = destination.check(config);
+    assertTrue(actual.getMessage().contains("State code: 08001;"));
+  }
+
+  @Test
+  public void testCheckIncorrectPortFailure() {
+    var config = buildConfigNoJdbcParameters();
+    ((ObjectNode) config).put(JdbcUtils.PORT_KEY, "30000");
+    var destination = new PostgresDestination();
+    final AirbyteConnectionStatus actual = destination.check(config);
+    assertTrue(actual.getMessage().contains("State code: 08001;"));
+  }
+
+  @Test
+  public void testCheckIncorrectDataBaseFailure() {
+    var config = buildConfigNoJdbcParameters();
+    ((ObjectNode) config).put(JdbcUtils.DATABASE_KEY, "wrongdatabase");
+    var destination = new PostgresDestination();
+    final AirbyteConnectionStatus actual = destination.check(config);
+    assertTrue(actual.getMessage().contains("State code: 3D000;"));
+  }
+
+  @Test
+  public void testUserHasNoPermissionToDataBase() throws Exception {
+    final JdbcDatabase database = PostgreSQLContainerHelper.getJdbcDatabaseFromConfig(PostgreSQLContainerHelper.getDataSourceFromConfig(config));
+
+    database.execute(connection -> connection.createStatement()
+        .execute(String.format("create user %s with password '%s';", USERNAME, PASSWORD)));
+    database.execute(connection -> connection.createStatement()
+        .execute(String.format("create database %s;", DATABASE)));
+    // deny access for database for all users from group public
+    database.execute(connection -> connection.createStatement()
+        .execute(String.format("revoke all on database %s from public;", DATABASE)));
+
+    ((ObjectNode) config).put(JdbcUtils.USERNAME_KEY, USERNAME);
+    ((ObjectNode) config).put(JdbcUtils.PASSWORD_KEY, PASSWORD);
+    ((ObjectNode) config).put(JdbcUtils.DATABASE_KEY, DATABASE);
+
+    var destination = new PostgresDestination();
```

Comment:
Can we keep consistency with naming similar to `PostgresDestinationTest#sanityTest`? The test there uses 
```java
final Destination destination = new PostgresDestionation()
```

Reply:
updated it